lib/robots/dor_repo/release/update_marc.rb

```diff
@@ -13,8 +13,33 @@ def initialize
         #
         # @param [String] druid -- the Druid identifier for the object to process
         def perform_work
-          logger.debug "update_marc working on #{druid}"
-          object_client.update_marc_record
+          logger.debug "update_marc handing off to dor-services-app job for update to #{druid}"
+          update_marc_record
+
+          # Since the actual update work is completed by dor-services-app in a job,
+          # workflow logging (marking step completed or failed) is done there.
+          LyberCore::ReturnState.new(status: :noop, note: 'Initiated update_marc_record API call.')
```

Comment:
i copied it from the validate-moab step in pres bots, thinking that was what kept the wf step open, to be marked as completed or failed by the code that actually does the work (which in the case of that example is handled by pres cat).  i think that also might make for more honest reporting of length of time to complete the step?  @ndushay might know better -- iirc she did the validate-moab implementation on both the pres cat and pres bots side of that wf step.

while it's prob not a huge deal either way, i would definitely vote for not marking it complete before it actually is.  i do think there are places where we do that, but can't recall offhand whether that's led to trouble or odd behavior.

Reply:
DSA doesn't update the workflow step as part of updating the MARC, so this isn't like validate moab.

Reply:
ah, sorry, my unstated assumption based on pairing with @lwrubel last week was that she'd add that part to the new `UpdateMarcJob` she's working on, doing something similar to [the workflow step update code in `ValidateMoabJob`](https://github.com/sul-dlss/preservation_catalog/blob/f67e26980762f3460420e1181af3067178420669/app/jobs/validate_moab_job.rb#L88-L118).

Reply:
It seems to me that the main benefits of not immediately marking it `completed` are:
1. Surfacing any errors for the workflow step in workflow grid.  Alternatively, if the step were immediately marked `completed`, failure could be reported to Honeybadger, and Sidekiq could handle retries.  @andrewjbtw, do you have a preference on this front?
2. Honest reporting of elapsed time for the update, including possible retries (both for the update itself and for the refetching of the record to confirm that the update has landed).  Being able to easily query that seems like it could be useful for debugging, esp if things pile up at this step, or if we see a lot of errors on this step.  Perhaps this would be easy enough to figure out from the logs, though maybe only for the log retention period?
3. Not marking it `completed` till the update is confirmed to have landed keeps this (from the perspective of when workflow steps start) a synchronous/blocking operation, which i think it is currently?  And if so, that'd keep a bit more similarity to the way the workflow executes currently.  Though, to be fair, I don't know one way or the other if any dowstream steps assume this workflow steps assume completion of `update-marc`.  Maybe another question for @andrewjbtw?

Reply:
`update-marc` is the last step in the `releaseWF` and the sole purpose of the `releaseWF` is to complete whatever work is needed to get things to appear in SW. What would be the benefit of marking the step completed when it isn't? This would have the effect of making the `releaseWF` looks like it finished successfully when it hadn't.

I look at the workflow grid to check whether steps have failed or are marked as running for an unusually long time. If there's a problem, I use the workflow-related facets to group together errored items for further analysis, retries, etc. If the workflow grid isn't reliable, debugging and remediation at a batch level is much harder, usually involving rake tasks or the rails console, and introducing more risk into each update. (Often that's required anyway, but it depends.) There are a lot of signs that under FOLIO this step is going to run into issues and I'd like to avoid spelunking in the depths of HB and sidekiq to find things that might not have worked.

Reply:
from lyber-core/robot.rb. 

```
# noop allows a robot to not set a workflow as complete, e.g., if that is delegated to another service.
```

and we seem to do this for the shelve step (or at least have a test saying we do), and in the publish step ... (https://github.com/search?q=org%3Asul-dlss+%3Anoop&type=code )

Reply:
Both of those steps remain at "started" while they're running, or at least they don't immediately go to "completed", so that's probably working as intended. They both have retries enabled, which I think has helped us more often than it's led to infinite retries (though publish has done that on more than one occasion).

Reply:
I guess I don't understand what has motivated the change from the existing approach for this workflow step. Changing it is not required by the changes to the update marc service. In fact, this entire PR is unnecessary, so why add additional complications to this already fraught area of work?

I'd suggest that if you get update marc working and the workflow state handling is not working as desired then change it, but leave it the way it is for now.

Reply:
Question withdrawn after conversation with @lwrubel.

Reply:
Tested in QA with the UpdateMarcJob communicating back via LogSuccessJob / LogFailureJob. Workflow step should show as active until it completes (or fails, although I wasn't able to test that in QA).